app/models/manageiq/providers/redhat/infra_manager/ovirt_services/strategies/v3.rb

```diff
@@ -17,5 +25,302 @@ def cluster_name_href(href)
+      vm.with_provider_object(&:status)[:state] == "down"
+    end
+
+    def powered_on_in_provider?(vm)
+      vm.with_provider_object(&:status)[:state] == "up"
+    end
+
+    def vm_boot_from_cdrom(operation, name)
+      operation.get_provider_destination.boot_from_cdrom(name)
+    rescue Ovirt::VmNotReadyToBoot
+      raise OvirtServices::VmNotReadyToBoot
+    end
+
+    def vm_boot_from_network(operation)
+      operation.get_provider_destination.boot_from_network
+    rescue Ovirt::VmNotReadyToBoot
+      raise OvirtServices::VmNotReadyToBoot
+    end
+
+    def get_template_proxy(template, connection)
+      connection ||= ext_management_system.connect
+      template_proxy = connection.get_resource_by_ems_ref(template.ems_ref)
+      GeneralUpdateMethodNamesDecorator.new(template_proxy)
+    end
+
+    def get_vm_proxy(vm, connection)
+      connection ||= ext_management_system.connect
+      vm_proxy = connection.get_resource_by_ems_ref(vm.ems_ref)
+      GeneralUpdateMethodNamesDecorator.new(vm_proxy)
+    end
+
+    def collect_disks_by_hrefs(disks)
+      vm_disks = []
+
+      ext_management_system.try(:with_provider_connection) do |rhevm|
+        disks.each do |disk|
+          begin
+            vm_disks << Ovirt::Disk.find_by_href(rhevm, disk)
+          rescue Ovirt::MissingResourceError
+            nil
+          end
+        end
+      end
+      vm_disks
+    end
+
+    def shutdown_guest(operation)
+      operation.with_provider_object(&:shutdown)
+    rescue Ovirt::VmIsNotRunning
+    end
+
+    def start_clone(source, clone_options, phase_context)
+      source.with_provider_object do |rhevm_template|
+        vm = rhevm_template.create_vm(clone_options)
+        populate_phase_context(phase_context, vm)
+      end
+    end
+
+    def vm_start(vm, cloud_init)
+      vm.with_provider_object do |rhevm_vm|
+        rhevm_vm.start { |action| action.use_cloud_init(true) if cloud_init }
+      end
+    rescue Ovirt::VmAlreadyRunning
+    end
+
+    def vm_stop(vm)
+      vm.with_provider_object(&:stop)
+    rescue Ovirt::VmIsNotRunning
+    end
+
+    def vm_suspend(vm)
+      vm.with_provider_object(&:suspend)
+    end
+
+    def vm_reconfigure(vm, options = {})
+      log_header = "EMS: [#{ext_management_system.name}] #{vm.class.name}: id [#{vm.id}], name [#{vm.name}], ems_ref [#{vm.ems_ref}]"
+      spec = options[:spec]
+
+      _log.info("#{log_header} Started...")
+
+      vm.with_provider_object do |rhevm_vm|
+        update_vm_memory(rhevm_vm, spec["memoryMB"] * 1.megabyte) if spec["memoryMB"]
+
+        cpu_options = {}
+        cpu_options[:cores] = spec["numCoresPerSocket"] if spec["numCoresPerSocket"]
+        cpu_options[:sockets] = spec["numCPUs"] / (cpu_options[:cores] || vm.cpu_cores_per_socket) if spec["numCPUs"]
+
+        rhevm_vm.cpu_topology = cpu_options if cpu_options.present?
+      end
+
+      _log.info("#{log_header} Completed.")
+    end
+
+    class NicsDecorator < SimpleDelegator
```

Comment:
I would like to do this in future refactoring.
There are several more refactoring steps on this code and this will be one of them, but currently will be just a bit easier to do while the classes are still here.